File: app/src/main/java/com/rdd/finy/data/Wallet.java
package com.rdd.finy.data;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.PrimaryKey;

@Entity
public class Wallet {

    @PrimaryKey(autoGenerate = true)
    private int id;

    private String title;

    private int budjet;

    private int colorId;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getBudjet() {
        return budjet;
    }

    public void setBudjet(int budjet) {
        this.budjet = budjet;
    }

    public int getColorId() {
        return colorId;
    }

    public void setColorId(int colorId) {
        this.colorId = colorId;
    }
}

File: app/src/main/java/com/rdd/finy/data/Wallet.kt
package com.rdd.finy.data

import android.graphics.Color
import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

@Entity
class Wallet(@PrimaryKey(autoGenerate = true)
              var id: Long = 0,
             var title: String = "",
             var currency: String = Currency.getInstance(Locale.US).displayName,
             var balance: Double = 0.0,
             var backColor: Int = Color.MAGENTA,
             var bottomDivider: Double = 0.0,
             var upperDivider: Double = -1.0,
             var isActive: Boolean = true)
{

    val hasUpperDivider : Boolean
    get() {
        return upperDivider >= 0.0
    }

    val hasBottomDivider: Boolean
    get(){
        return bottomDivider > 0.0
    }

    val couldBeSaved: Boolean
    get(){
        if(upperDivider<balance && hasUpperDivider)return false
        if(bottomDivider>balance && hasBottomDivider)return false
        return true
    }
}